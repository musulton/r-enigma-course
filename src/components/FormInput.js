import React from "react";
import {FormControl, FormGroup, FormLabel, Form} from "react-bootstrap";

const isFileOrNotObject = (value, type) => type === "file" && (typeof value !== "object")

const FormInput = (props) => {
    const {label, type, placeholder, value, onChange, disabled} = props
    let formProps;

    switch (type) {
        case "textarea":
            formProps = { as: type, value }
            break;
        case "file":
            formProps = { type }
            break;
        default:
            formProps = { type, value }
            break;
    }

    return (
        <FormGroup className="mb-3">
            <FormLabel>{label}</FormLabel>
            <FormControl
                {...formProps}
                placeholder={placeholder}
                onChange={onChange}
                disabled={disabled}
            />
            {isFileOrNotObject(value, type) && <Form.Text>{value}</Form.Text>}
        </FormGroup>
    )
}

export default FormInput
