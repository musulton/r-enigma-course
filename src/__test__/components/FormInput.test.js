import FormInput from "../../components/FormInput";
import {render} from "@testing-library/react";

describe("FormInput component", () => {
    const mockProps = {
        label: "Label",
        placeholder: "Placeholder",
        value: "value",
        onChange: jest.fn(),
        disabled: false
    }

    test("should render textarea type correctly", () => {
        const {asFragment} = render(<FormInput {...mockProps} type="textarea" />)

        expect(asFragment()).toMatchSnapshot()
    })

    test("should render file type correctly", () => {
        const {asFragment} = render(<FormInput {...mockProps} type="file" />)

        expect(asFragment()).toMatchSnapshot()
    })

    test("should render text type correctly", () => {
        const {asFragment} = render(<FormInput {...mockProps} type="text" />)

        expect(asFragment()).toMatchSnapshot()
    })
})
