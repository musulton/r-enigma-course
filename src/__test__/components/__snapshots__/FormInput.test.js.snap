// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput component should render file type correctly 1`] = `
<DocumentFragment>
  <div
    class="mb-3"
  >
    <label
      class="form-label"
    >
      Label
    </label>
    <input
      class="form-control"
      placeholder="Placeholder"
      type="file"
    />
    <small
      class="form-text"
    >
      value
    </small>
  </div>
</DocumentFragment>
`;

exports[`FormInput component should render text type correctly 1`] = `
<DocumentFragment>
  <div
    class="mb-3"
  >
    <label
      class="form-label"
    >
      Label
    </label>
    <input
      class="form-control"
      placeholder="Placeholder"
      type="text"
      value="value"
    />
  </div>
</DocumentFragment>
`;

exports[`FormInput component should render textarea type correctly 1`] = `
<DocumentFragment>
  <div
    class="mb-3"
  >
    <label
      class="form-label"
    >
      Label
    </label>
    <textarea
      class="form-control"
      placeholder="Placeholder"
    >
      value
    </textarea>
  </div>
</DocumentFragment>
`;
